cmd/dex-worker/main.go

```diff
@@ -48,6 +48,7 @@ func main() {
 
 	enableRegistration := fs.Bool("enable-registration", false, "Allows users to self-register")
 	enableClientRegistration := fs.Bool("enable-client-registration", false, "Allow dynamic registration of clients")
+	registerOnFirstLogin := fs.Bool("register-on-first-login", false, "When a user logs in through a federated identity service, automatically register them if they don't have an account. This flag is independent of enable-registration.")
```

Comment:
> What happens if enableRegistration=true and registerOnFirstLogin=true?

You'd be able to use the registration page (and the registration endpoint), as well as be automatically registered via logging in through a remote connector.